### ISSUE TITLE
Conversion from `IntPtr` to `nint` to `nuint` incorrectly reported as redundant.

### ISSUE THREAD

**Issue by tannergooding** [dotnet/roslyn#47800]
The following code incorrectly reports a cast from `IntPtr` -> `nint` -> `nuint` as redundant:
```csharp
using System;

public class C {
    public nuint M(nint x) => (nuint)x;

    public nint M(nuint x) => (nint)x;

    public nuint N(IntPtr x) => (nuint)(nint)x; // IDE0004: Cast is redundant

    public nint N(UIntPtr x) => (nint)(nuint)x; // IDE0004: Cast is redundant

    public nuint O(IntPtr x) => (nuint)x; // CS0030: Cannot convert type 'System.IntPtr' to 'nuint'

    public nint O(UIntPtr x) => (nint)x;  // CS0030: Cannot convert type 'System.UIntPtr' to 'nint'
}
```

However, it is required as casting directly from `IntPtr` to `nuint` is disallowed (only a conversion from `nint` to `nuint` is allowed)

Sharplab for master from Sep 15th:
https://sharplab.io/#v2:EYLgtghgzgLgpgJwDQxASwDYB8ACAmARgFgAoUnAZgAJ8qBhKgb1KtZuoDsBXNDmKgLIAKDr34APAJRUAvAD4qInn0niA3KRZtKVUX0FKxVKbIUixqjWRJt2u5fwByQgJJ8ACjATHp8xdwtzFXUqAHpQqhcAEQBRAAYEgBYQemh+NCgqBDgAEy4OHIg+TRttTiNnAFU3GE9vEz8gmElDYLUwiOj4pJS6NKoMrNz8wuLrWx0A/QB5Vw8vH1N/B0sO+gBlBIo43qKOAHt+AGN9jgA3RH4YAE8ABzgqAHIcAgA6GrrHqhh9p6mYR4lCblGZCarzeq+Mx6ZohNZ0TZxba7DgHY6nC4IK53B7PN7g2peL4/P5iQEkAC+QA===

**Comment by tannergooding** [comment]
CC. @cston, I'm pretty sure this is an IDE issue, unless the conversion from `IntPtr->nuint` should be allowed.
